runelite-client/src/main/java/net/runelite/client/plugins/twitch/TwitchPlugin.java

```diff
@@ -0,0 +1,421 @@
+		{
+			subscribed = true;
+			if (originalMessage.contains("subscriber=1"))
+			{
+				if (originalMessage.contains("subscriber/0"))
+				{
+					badges = "<img=9>";
+				}
+				else if (originalMessage.contains("subscriber/3"))
+				{
+					badges = "<img=8>";
+				}
+				else if (originalMessage.contains("subscriber/6"))
+				{
+					badges = "<img=7>";
+				}
+				else if (originalMessage.contains("subscriber/12"))
+				{
+					badges = "<img=6>";
+				}
+				else if (originalMessage.contains("subscriber/24"))
+				{
+					badges = "<img=5>";
+				}
+				else if (originalMessage.contains("subscriber/48"))
+				{
+					badges = "<img=4>";
+				}
+			}
+		}
+		else
+		{
+			subscribed = false;
+		}
+
+		//Finding if the user is a moderator or broadcaster which allows them to talk during emote-only and sub-only while not actually subscribed
+		if (originalMessage.contains("mod=1") && originalMessage.toLowerCase().contains("display-name=" + config.channelName()))
+		{
+			mod = true;
+			badges = "<img=26>" + badges;
+		}
+		else
+		{
+			mod = false;
+		}
+
+		if (originalMessage.contains("value=broadcaster/1") && originalMessage.toLowerCase().contains("display-name=" + config.channelName()))
+		{
+			broadcaster = true;
+			badges = "<img=27>" + badges;
+		}
+		else
+		{
+			broadcaster = false;
+		}
+	}
+
+	/**
+	 * Triggers when the room state changes. I.E. Sub mode, slow mode, etc.
+	 * @param r
+	 */
+	@Handler
+	private void onRoomStateEvent(RoomStateEvent r)
+	{
+		String originalMessage = r.getOriginalMessage().toString();
+		//Checking if the subs-only state has changed
+		subsOnly = originalMessage.contains("subs-only=1");
+
+		//Checking if the emote-only state has changed
+		emoteOnly = originalMessage.contains("emote-only=1");
+	}
+
+	/**
+	 * Triggers when a user subscribes/re-subscribes
+	 * @param u
+	 */
+	@Handler
+	private void onUserNoticeEvent(UserNoticeEvent u)
+	{
+		if (!client.getGameState().equals(GameState.LOGGED_IN) || !config.subNotifs())
+		{
+			return;
+		}
+
+		String tColor = Integer.toHexString(config.subsTColor().getRGB() & 0xFFFFFF);
+		String oColor = Integer.toHexString(config.subsOColor().getRGB() & 0xFFFFFF);
+
+		String newSubMessage;
+		String originalMessage = u.getOriginalMessage().toString();
+		String[] subMessage = originalMessage.split(";");
+
+		//Fixes and issue where gifted subs say "Notification: msg-param-sub-plan-name=1000" instead of the proper sub message
+		if (subMessage[13].contains("msg-param-sub-plan-name="))
+		{
+			newSubMessage = subMessage[17].replace("\\s", " ").replace("system-msg=", "");
+		}
+		else
+		{
+			newSubMessage = subMessage[13].replace("\\s", " ").replace("system-msg=", "");
+		}
+
+		//Sends the sub notification
+		if (isChatboxTransparent())
+		{
+			client.addChatMessage(ChatMessageType.TRADE, "",
+				"<col=ffffff></col>[<col=9070ff>Twitch</col>] <col=" + tColor + ">" + newSubMessage + "</col>", "");
```

Comment:
Same as above, use constants and potentially pass this off to a method just to improve readability